src/unlock/unlocker.rs

```diff
@@ -224,187 +224,205 @@ impl From<Box<dyn Signer>> for AcpUnlocker {
         AcpUnlocker::new(AcpScriptSigner::new(signer))
     }
 }
-impl ScriptUnlocker for AcpUnlocker {
-    fn match_args(&self, args: &[u8]) -> bool {
-        self.signer.match_args(args)
-    }
 
-    fn is_unlocked(
-        &self,
-        tx: &TransactionView,
-        script_group: &ScriptGroup,
-        tx_dep_provider: &dyn TransactionDependencyProvider,
-    ) -> Result<bool, UnlockError> {
-        const POW10: [u64; 20] = [
-            1,
-            10,
-            100,
-            1000,
-            10000,
-            100000,
-            1000000,
-            10000000,
-            100000000,
-            1000000000,
-            10000000000,
-            100000000000,
-            1000000000000,
-            10000000000000,
-            100000000000000,
-            1000000000000000,
-            10000000000000000,
-            100000000000000000,
-            1000000000000000000,
-            10000000000000000000,
-        ];
-        let script_args = script_group.script.args().raw_data();
-        let min_ckb_amount = if script_args.len() > 20 {
-            let idx = script_args.as_ref()[20];
-            if idx >= 20 {
-                return Err(UnlockError::Other(format!("invalid min ckb amount config in script.args, got: {}, expected: value >=0 and value < 20", idx).into()));
-            }
-            POW10[idx as usize]
-        } else {
-            0
-        };
-        let min_udt_amount = if script_args.len() > 21 {
-            let idx = script_args.as_ref()[21];
-            if idx >= 39 {
-                return Err(UnlockError::Other(format!("invalid min udt amount config in script.args, got: {}, expected: value >=0 and value < 39", idx).into()));
-            }
-            if idx >= 20 {
-                (POW10[19] as u128) * (POW10[idx as usize - 19] as u128)
-            } else {
-                POW10[idx as usize] as u128
-            }
+fn acp_is_unlocked(
+    tx: &TransactionView,
+    script_group: &ScriptGroup,
+    tx_dep_provider: &dyn TransactionDependencyProvider,
+    script_args: &[u8],
```

Comment:
This is actually not script args, it's extra args of acp lock.
```suggestion
    acp_args: &[u8],
```